lib/active_job/queue_adapters/active_elastic_job_adapter.rb

```diff
@@ -137,58 +137,55 @@ def build_message(queue_name, serialized_job, timestamp)
             message_attributes: build_message_attributes(serialized_job)
           }
 
-          if queue_name.split('.').last == 'fifo'
-            args.merge!(fifo_required_params(serialized_job))
-          end
+          args.merge!(fifo_required_params(serialized_job)) if queue_name.split('.').last == 'fifo'
 
-          return args
+          args
         end
 
         def build_message_attributes(serialized_job)
           {
-            "message-digest".freeze => {
+            'message-digest'.freeze => {
               string_value: message_digest(serialized_job),
-              data_type: "String".freeze
+              data_type: 'String'.freeze
             },
             origin: {
               string_value: ActiveElasticJob::ACRONYM,
-              data_type: "String".freeze
+              data_type: 'String'.freeze
             }
           }
         end
 
         def fifo_required_params(serialized_job)
           parsed_job = JSON.parse(serialized_job)
+          deduplication_id = OpenSSL::Digest::MD5.hexdigest(parsed_job['job_class'] + parsed_job['arguments'].to_s)
```

Comment:
basically this is the change and the motivation of this PR, I think the job_id is a random uuid and the deduplication id should be something deterministic like my suggestion.